src/analysis/calculus/fderiv.lean

```diff
@@ -1907,32 +1907,27 @@ namespace continuous_linear_equiv
 
 In this section we establish that the set of continuous linear equivalences between two Banach
 spaces is an open subset of the space of linear maps between them.  These facts are placed here
-because the proof uses `is_bounded_bilinear_map.continuous`, proved just above as a consequence
+because the proof uses `is_bounded_bilinear_map.continuous_left`, proved just above as a consequence
 of its differentiability.
 -/
 
 protected lemma is_open [complete_space E] : is_open (range (coe : (E ≃L[𝕜] F) → (E →L[𝕜] F))) :=
 begin
+  nontriviality E,
   rw [is_open_iff_mem_nhds, forall_range_iff],
   refine λ e, mem_nhds_sets _ (mem_range_self _),
   let O : (E →L[𝕜] F) → (E →L[𝕜] E) := λ f, (e.symm : F →L[𝕜] E).comp f,
-  cases subsingleton_or_nontrivial E with _i _i; resetI,
-  { exact is_open_discrete _ },
-  have h_O : continuous O,
-  { have h_e_symm : continuous (λ (x : E →L[𝕜] F), (e.symm : F →L[𝕜] E)) := continuous_const,
-    exact is_bounded_bilinear_map_comp.continuous.comp (continuous_id.prod_mk h_e_symm) },
+  have h_O : continuous O := is_bounded_bilinear_map_comp.continuous_left,
   convert units.is_open.preimage h_O using 1,
   ext f',
   split,
   { rintros ⟨e', rfl⟩,
-    let w : units (E →L[𝕜] E) := continuous_linear_equiv.to_unit (e'.trans e.symm),
-    exact ⟨w, rfl⟩ },
+    exact ⟨continuous_linear_equiv.to_unit (e'.trans e.symm), rfl⟩ },
   { rintros ⟨w, hw⟩,
-    let e' : E ≃L[𝕜] E := continuous_linear_equiv.of_unit w,
-    use e'.trans e,
+    use (continuous_linear_equiv.of_unit w).trans e,
     ext x,
-    have he'w : e' x = w x := rfl,
-    simp [hw, he'w] }
+    have hw' : (continuous_linear_equiv.of_unit w) x = w x := rfl,
```

Comment:
Those parentheses shouldn't be needed. This line and the following one sound like we are missing something like
```lean
@[simp] lemma of_unit_apply (f : units (M →L[R] M)) (x : M) : of_unit f x = f x := rfl
```
right after the definition of `of_unit` so that `hw'` can be left out.

Reply:
It's funny, the simp-lemma already existed, but I forgot that I wrote it for the bundled equivalences rather than the bare maps, so it wasn't picking up.

Rewrote to use that simp-lemma, and also fixed the same parentheses error which I discovered there!  Thanks!